java-squid/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -0,0 +1,305 @@
+          final ExpressionTree targetExpression = ((MemberSelectExpressionTree) methodSelect).expression();
+          if (targetExpression.is(Tree.Kind.IDENTIFIER)) {
+            final IdentifierTree identifier = (IdentifierTree) targetExpression;
+            final SymbolicValue target = programState.getValue(identifier.symbol());
+            programState = closeResource(programState, target);
+          }
+        }
+      } else {
+        final List<SymbolicValue> values = programState.peekValues(syntaxNode.arguments().size() + 1);
+        final List<SymbolicValue> argumentValues = values.subList(1, values.size());
+        for (SymbolicValue target : argumentValues) {
+          programState = closeResource(programState, target);
+        }
+      }
+    }
+
+    private static boolean isOpeningResource(NewClassTree syntaxNode) {
+      if (isWithinTryHeader(syntaxNode)) {
+        return false;
+      }
+      final Type type = syntaxNode.symbolType();
+      return needsClosing(type);
+    }
+
+    private static boolean isWithinTryHeader(NewClassTree syntaxNode) {
+      final Tree parent = syntaxNode.parent();
+      if (parent != null) {
+        final Tree grandParent = parent.parent();
+        if (grandParent != null) {
+          final Tree greatGrandParent = grandParent.parent();
+          return greatGrandParent != null && greatGrandParent.is(Tree.Kind.TRY_STATEMENT);
+        }
+      }
+      return false;
+    }
+
+    private static CloseableArgument getCloseableArgument(NewClassTree syntaxNode, List<SymbolicValue> arguments) {
+      Iterator<SymbolicValue> iterator = arguments.iterator();
+      for (ExpressionTree argument : syntaxNode.arguments()) {
+        if (!iterator.hasNext()) {
+          throw new IllegalStateException("Mismatch between declared constructor arguments and argument values!");
+        }
+        final Type type = argument.symbolType();
+        final SymbolicValue value = iterator.next();
+        if (isCloseable(type)) {
+          return new CloseableArgument(needsClosing(type) ? value : null);
+        }
+      }
+      return new CloseableArgument(null);
+    }
+
+    private static ProgramState closeResource(ProgramState programState, final SymbolicValue target) {
+      if (target != null && isOpened(programState, target)) {
+        return programState.addConstraint(target.wrappedValue(), NullConstraint.CLOSED);
+      }
+      return programState;
+    }
+
+    private static boolean isOpened(ProgramState programState, SymbolicValue value) {
+      return NullConstraint.OPENED.equals(programState.getConstraint(value.wrappedValue()));
+    }
+
+    private static boolean isClosingResource(Symbol constructor) {
+      return constructor.isMethodSymbol() && CLOSE_METHOD_NAME.equals(constructor.name());
+    }
+  }
+
+  // TODO This variable should be placed into the checker context to ensure that we always get the value corresponding to the previous post
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1135)